test/Microsoft.AspNetCore.Mvc.RazorPages.Test/Infrastructure/PageDirectiveFeatureTest.cs

```diff
@@ -15,186 +19,141 @@ public class PageDirectiveFeatureTest
+                });
         }
 
         [Fact]
-        public void TryGetPageDirective_NoQuotesAroundPath_IsNotTemplate()
+        public void TryGetPageDirective_NoQuotesAroundPath_LogsWarning()
         {
             // Arrange
-            string template;
+            var expected = "The page directive at 'Test.cshtml' is malformed. Please fix the following issues: The 'page' directive expects a string surrounded by double quotes.";
+            var sink = new TestSink();
+            var logger = new TestLogger("logger", sink, enabled: true);
             var projectItem = new TestRazorProjectItem(@"@page Some/Path/{value}");
 
             // Act & Assert
-            Assert.False(PageDirectiveFeature.TryGetPageDirective(projectItem, out template));
+            Assert.True(PageDirectiveFeature.TryGetPageDirective(logger, projectItem, out var template));
             Assert.Null(template);
-        }
-
-        [Fact]
-        public void TryGetPageDirective_WrongNewLine()
-        {
-            // Arrange
-            var wrongNewLine = Environment.NewLine == "\r\n" ? "\n" : "\r\n";
-
-            string template;
-            var projectItem = new TestRazorProjectItem($"@page \"Some/Path/{{value}}\" {wrongNewLine}");
-
-            // Act & Assert
-            Assert.True(PageDirectiveFeature.TryGetPageDirective(projectItem, out template));
-            Assert.Equal("Some/Path/{value}", template);
+            var logs = sink.Writes.Select(w => w.State.ToString().Trim()).ToList();
+            Assert.Collection(sink.Writes,
+                log =>
+                {
+                    Assert.Equal(LogLevel.Warning, log.LogLevel);
+                    Assert.Equal(expected, log.State.ToString());
+                });
         }
 
         [Fact]
         public void TryGetPageDirective_NewLineBeforeDirective()
         {
             // Arrange
-            string template;
             var projectItem = new TestRazorProjectItem("\n @page \"Some/Path/{value}\"");
+            var sink = new TestSink();
+            var logger = new TestLogger("logger", sink, enabled: true);
 
             // Act
-            Assert.True(PageDirectiveFeature.TryGetPageDirective(projectItem, out template));
-            Assert.Equal("Some/Path/{value}", template);
-        }
-
-        [Fact]
-        public void TryGetPageDirective_WhitespaceBeforeDirective()
-        {
-            // Arrange
-            string template;
-            var projectItem = new TestRazorProjectItem(@"   @page ""Some/Path/{value}""
-");
-
-            // Act & Assert
-            Assert.True(PageDirectiveFeature.TryGetPageDirective(projectItem, out template));
+            Assert.True(PageDirectiveFeature.TryGetPageDirective(logger, projectItem, out var template));
             Assert.Equal("Some/Path/{value}", template);
-        }
-
-        [Fact]
-        public void TryGetPageDirective_JunkBeforeNewline()
```

Comment:
These are just testing parser behavior. We shouldn't need separate tests for it here.